File: web/src/api/platform.ts
import request from "./request";
import type { GitPlatformConfig } from "../types";

export const platformApi = {
  // Get platform config
  getConfig: () => {
    return request.get<GitPlatformConfig | { exists: false; message: string }>(
      "/platform/config"
    );
  },
  // Update platform config
  updateConfig: (data: GitPlatformConfig) => {
    return request.put<{
      message: string;
      config?: GitPlatformConfig;
    }>("/platform/config", data);
  },

  // Test connection (accepts configuration data)
  testConnection: (data: {
    platform_type: string;
    base_url: string;
    access_token: string;
  }) => {
    return request.post<{ success: boolean; message: string }>(
      "/platform/test",
      data
    );
  },
};

File: internal/api/dto/platform.go
package dto

// UpdatePlatformConfigRequest represents the request body for updating platform config
type UpdatePlatformConfigRequest struct {
	PlatformType string  `json:"platform_type" binding:"required"`
	BaseURL      string  `json:"base_url" binding:"required"`
	AccessToken  *string `json:"access_token"` // Pointer: nil = keep existing, non-nil = update
	IsActive     bool    `json:"is_active"`
}

// TestConnectionRequest represents the request body for testing platform connection
// This allows users to test configuration before saving
type TestConnectionRequest struct {
	PlatformType string `json:"platform_type" binding:"required"`
	BaseURL      string `json:"base_url" binding:"required"`
	AccessToken  string `json:"access_token" binding:"required"`
}
